### ISSUE TITLE
convert_from_transformers

### ISSUE THREAD

**Issue by dkamalakar** [deepset-ai/FARM#566]
Having little bit trouble following what happens behind the scene for convert_from_transformers

I am trying to develope a QA model using FARMReader and since I have the huggigface model loaded from local, have used conver_from_transformers as follows

tokenizer = Tokenizer.load(pretrained_model_name_or_path = path, tokenizer_class="BertTokenizer")
processor = InferenceProcessor(tokenizer=tokenizer, max_seq_len = 384)
model = AdaptiveMode.convert_from_transformers(model_name_or_path = path, device = "cuda", task_type="embeddings")

and saved the model 

processor.save (save_dir = target_dir)
model.save(save_dir=target_dir)

BUT when I try to use the FARMReader 

reader = FARMReader(model_name_or_path=target_dir, use_gpu=True)

I get error message as below

**_Updating with additional information:_**
I was actually getting index 0 out of range , but when I use task_type as "question_answering" for 
model = AdaptiveMode.convert_from_transformers(model_name_or_path = path, device = "cuda", task_type="question_answering")

I get key_error "question_answering"

If I use task_type as "embeddings" for AdaptiveModel.convert_from_transformers, i dont think it generates prediction_head and hence the index 0 error. But when I use task_type as "question_answering" I get above error.
I am using bert_large_cased

Any suggestion will be greatly appreciated

**Comment by Timoeller** [comment]
Hey I am unsure about what you mean, so let me recap what you did:
- You have a QA model that you trained with huggingface and saved it loacally at `path` 
- Then you load it in FARM, save it as a FARM model
- Then the error occurs when using the FARMReader for loading the model again

So there should be a FARM style QAmodel in `target_dir`. Could you write which files are present in this dir and what are the contents of prediction head config (if there is any).


---------------
We also updated the conversion in #576 today. Maybe it is worth checking out the updated code and examples?

**Comment by dkamalakar** [comment]
Thank you for your response

I have not trained the Huggingface model. Using pre-trained
'bert-large-uncased-whole-world-masking-finetuned-squad'

I am getting error in step 3 you mentioned. The target dir contains
processor_config and prediction heads

Just to provide some additional information
I am trying to use DensePassageRetriever for Q&A on bunch of documents.

The steps I am following
1. Retriever using pre-trained query and passage embedding model. Using
FAISSDocumentStore as document store
2. Using AdaptiveModel.conver_from_transformers() with huggingface model
referenced above. I am not training or finetuning the model. Using as is
from huggingface
3. Trying to use FARMReader to create reader...

Getting error message mentioned above.

Thank you in advance for your help

**Comment by Timoeller** [comment]
Ok so this is then purely haystack related.

Please open issues in haystack if you have questions about the haystack framework.

About your problem, please just use the FARMReader directly on the model_string that corresponds to the huggingface model zoo, it should download and load the QA models correctly:
```
reader = FARMReader(model_name_or_path="bert-large-uncased-whole-word-masking-finetuned-squad", use_gpu=True)
```

btw you missspelled whole-world-masking in the model string, it is whole-word-masking and not obscuring the whole planet

**Comment by dkamalakar** [comment]
I have raised question in haystack forum. Unfortunately I cannot download directly due to firewall issues. Thanks for your reply

**Comment by Timoeller** [comment]
Understood. Firewall issues and loading the model locally is a bit more tricky and opening the issue here in FARM was a reasonable idea.

Though thanks for creating the issue in haystack, Tholors proposed solution there should fix your issue and will help other users as well - I bet many people run into this problem when using haystack.
